### PR TITLE
Test framework for tsv

### PR DESCRIPTION
**Pull request by ckairen**
DO NOT MERGE YET

### CODE REVIEW

File: eng/tools/TypeSpecValidation/test/test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { GitDiffRule } from "../src/rules/git-diff.js";
+import { TsvTestHost } from "./tsv-test-host.js";
+import {expect} from 'chai';
```
Comment (ckairen) [inline eng/tools/TypeSpecValidation/test/test.ts:3]: Use assert instead of chai

File: eng/tools/TypeSpecValidation/test/test.ts
Comment (mikeharder) [inline eng/tools/TypeSpecValidation/test/test.ts:1]: Let's use one test file per src file.  So the git-diff tests would live in file `git-diff.test.ts`.

File: eng/tools/TypeSpecValidation/test/tsv-test-host.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { TsvHost } from "../src/tsv-host.js";
+
+export const TsvTestHost: TsvHost = {
+    gitOperation: (folder: string) => {
+        return {
+            status: () => {return {modified:[`${folder}\n`], isClean: ()=>{return false}}},
```
Comment (mikeharder) [inline eng/tools/TypeSpecValidation/test/tsv-test-host.ts:6]: We will need some way to configure the behavior of these mocks for each test.  We could use constructor arguments or APIs after the mock is created.

**Review by mikeharder — CHANGES_REQUESTED** [review]
I like the overall direction of using a test host with manual mocks.

**Review by mikeharder — APPROVED** [review]

**Pull request merged**